`azurerm_sentinel_data_connector_microsoft_threat_intelligence` failed to import

### Is there an existing issue for this?

- [X] I have searched the existing issues

### Community Note

<!--- Please keep this note for the community --->

* Please vote on this issue by adding a :thumbsup: [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave comments along the lines of "+1", "me too" or "any updates", they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment and review the [contribution guide](https://github.com/hashicorp/terraform-provider-azurerm/blob/main/contributing/README.md) to help.

<!--- Thank you for keeping this note for the community --->


### Terraform Version

1.9.0

### AzureRM Provider Version

4.0.0

### Affected Resource(s)/Data Source(s)

azurerm_sentinel_data_connector_microsoft_threat_intelligence

### Terraform Configuration Files

```hcl
N/A
```


### Debug Output/Panic Output

```shell
stderr=
  | 2024-08-23T17:22:35.815+0800 [ERROR] terraform-provider-azurerm.terraform-provider-azurerm: [ERROR] setting state: Invalid address to set: []string{"bing_safety_phishing_url_lookback_date"}
  | 2024-08-23T17:22:35.815+0800 [ERROR] terraform-provider-azurerm.terraform-provider-azurerm: Response contains error diagnostic: diagnostic_severity=ERROR diagnostic_summary="setting \"bing_safety_phishing_url_lookback_date\": Invalid address to set: []string{\"bing_safety_phishing_url_lookback_date\"}" tf_req_id=99d6ba91-8872-ec9f-b566-16b8a77d5610 tf_rpc=ReadResource @module=sdk.proto diagnostic_detail="setting \"bing_safety_phishing_url_lookback_date\": Invalid address to set: []string{\"bing_safety_phishing_url_lookback_date\"}" tf_proto_version=5.6 tf_provider_addr=registry.terraform.io/hashicorp/azurerm tf_resource_type=azurerm_sentinel_data_connector_microsoft_threat_intelligence @caller=/home/magodo/github/terraform-provider-azurerm/vendor/github.com/hashicorp/terraform-plugin-go/tfprotov5/internal/diag/diagnostics.go:58 timestamp="2024-08-23T17:22:35.815+0800"
  | 2024-08-23T17:22:35.818+0800 [ERROR] terraform-provider-azurerm: setting "bing_safety_phishing_url_lookback_date": Invalid address to set: []string{"bing_safety_phishing_url_lookback_date"}: setting "bing_safety_phishing_url_lookback_date": Invalid address to set: []string{"bing_safety_phishing_url_lookback_date"}
```


### Expected Behaviour

_No response_

### Actual Behaviour

_No response_

### Steps to Reproduce

_No response_

### Important Factoids

_No response_

### References

_No response_

--- a/internal/services/sentinel/sentinel_data_connector_microsoft_threat_intelligence.go
+++ b/internal/services/sentinel/sentinel_data_connector_microsoft_threat_intelligence.go
@@ -11,7 +11,6 @@ import (
 
 	"github.com/hashicorp/go-azure-sdk/resource-manager/operationalinsights/2022-10-01/workspaces"
 	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
-	"github.com/hashicorp/terraform-provider-azurerm/internal/features"
 	"github.com/hashicorp/terraform-provider-azurerm/internal/sdk"
 	"github.com/hashicorp/terraform-provider-azurerm/internal/services/sentinel/parse"
 	"github.com/hashicorp/terraform-provider-azurerm/internal/services/sentinel/validate"
@@ -27,7 +26,6 @@ type DataConnectorMicrosoftThreatIntelligenceModel struct {
 	Name                                    string `tfschema:"name"`
 	WorkspaceId                             string `tfschema:"log_analytics_workspace_id"`
 	TenantId                                string `tfschema:"tenant_id"`
-	BingSafetyPhishingUrlLookBackDate       string `tfschema:"bing_safety_phishing_url_lookback_date"`
 	MicrosoftEmergingThreatFeedLookBackDate string `tfschema:"microsoft_emerging_threat_feed_lookback_date"`
 }
 
@@ -37,7 +35,7 @@ type DataConnectorMicrosoftThreatIntelligenceDataType struct {
 }
 
 func (s DataConnectorMicrosoftThreatIntelligenceResource) Arguments() map[string]*schema.Schema {
-	res := map[string]*schema.Schema{
+	return map[string]*schema.Schema{
 		"name": {
 			Type:         pluginsdk.TypeString,
 			Required:     true,
@@ -64,32 +62,10 @@ func (s DataConnectorMicrosoftThreatIntelligenceResource) Arguments() map[string
 		"microsoft_emerging_threat_feed_lookback_date": {
 			Type:         pluginsdk.TypeString,
 			ForceNew:     true,
-			Optional:     !features.FourPointOh(),
-			Required:     features.FourPointOh(),
+			Required:     true,
 			ValidateFunc: validation.IsRFC3339Time,
-			AtLeastOneOf: func() []string {
-				if !features.FourPointOh() {
-					return []string{"bing_safety_phishing_url_lookback_date", "microsoft_emerging_threat_feed_lookback_date"}
-				}
-				return []string{}
-			}(),
 		},
 	}
-
-	if !features.FourPointOh() {
-		// this has been removed in newer API version, and it's actually not working in current API version
-		// TODO Remove in 4.0
-		res["bing_safety_phishing_url_lookback_date"] = &schema.Schema{
-			Deprecated:   "This field is deprecated and will be removed in version 4.0 of the AzureRM Provider.",
-			Type:         pluginsdk.TypeString,
-			Optional:     true,
-			ForceNew:     true,
-			ValidateFunc: validation.IsRFC3339Time,
-			AtLeastOneOf: []string{"bing_safety_phishing_url_lookback_date", "microsoft_emerging_threat_feed_lookback_date"},
-		}
-	}
-
-	return res
 }
 
 func (s DataConnectorMicrosoftThreatIntelligenceResource) Attributes() map[string]*schema.Schema {
@@ -140,7 +116,6 @@ func (s DataConnectorMicrosoftThreatIntelligenceResource) Create() sdk.ResourceF
 				Kind: securityinsight.KindBasicDataConnectorKindMicrosoftThreatIntelligence,
 				MSTIDataConnectorProperties: &securityinsight.MSTIDataConnectorProperties{
 					DataTypes: &securityinsight.MSTIDataConnectorDataTypes{
-						BingSafetyPhishingURL:       expandSentinelDataConnectorMicrosoftThreatIntelligenceBingSafetyPhishingUrl(metaModel),
 						MicrosoftEmergingThreatFeed: expandSentinelDataConnectorMicrosoftThreatIntelligenceMicrosoftEmergingThreatFeed(metaModel),
 					},
 					TenantID: &tenantId,
@@ -193,14 +168,6 @@ func (s DataConnectorMicrosoftThreatIntelligenceResource) Read() sdk.ResourceFun
 			}
 
 			if dt := dc.DataTypes; dt != nil {
-				if dt.BingSafetyPhishingURL != nil {
-					if strings.EqualFold(string(dt.BingSafetyPhishingURL.State), string(securityinsight.DataTypeStateEnabled)) {
-						state.BingSafetyPhishingUrlLookBackDate, err = flattenSentinelDataConnectorMicrosoftThreatIntelligenceTime(*dt.BingSafetyPhishingURL.LookbackPeriod)
-						if err != nil {
-							return fmt.Errorf("flattening `bing_safety_phishing_url`: %+v", err)
-						}
-					}
-				}
 				if dt.MicrosoftEmergingThreatFeed != nil {
 					if strings.EqualFold(string(dt.MicrosoftEmergingThreatFeed.State), string(securityinsight.DataTypeStateEnabled)) {
 						state.MicrosoftEmergingThreatFeedLookBackDate, err = flattenSentinelDataConnectorMicrosoftThreatIntelligenceTime(*dt.MicrosoftEmergingThreatFeed.LookbackPeriod)
@@ -238,20 +205,6 @@ func (s DataConnectorMicrosoftThreatIntelligenceResource) Delete() sdk.ResourceF
 
 func (s DataConnectorMicrosoftThreatIntelligenceResource) IDValidationFunc() pluginsdk.SchemaValidateFunc {
 	return validate.DataConnectorID
-}
-
-func expandSentinelDataConnectorMicrosoftThreatIntelligenceBingSafetyPhishingUrl(input DataConnectorMicrosoftThreatIntelligenceModel) *securityinsight.MSTIDataConnectorDataTypesBingSafetyPhishingURL {
-	if input.BingSafetyPhishingUrlLookBackDate == "" {
-		return &securityinsight.MSTIDataConnectorDataTypesBingSafetyPhishingURL{
-			LookbackPeriod: utils.String(""),
-			State:          securityinsight.DataTypeStateDisabled,
-		}
-	}
-
-	return &securityinsight.MSTIDataConnectorDataTypesBingSafetyPhishingURL{
-		LookbackPeriod: utils.String(input.BingSafetyPhishingUrlLookBackDate),
-		State:          securityinsight.DataTypeStateEnabled,
-	}
 }
 
 func expandSentinelDataConnectorMicrosoftThreatIntelligenceMicrosoftEmergingThreatFeed(input DataConnectorMicrosoftThreatIntelligenceModel) *securityinsight.MSTIDataConnectorDataTypesMicrosoftEmergingThreatFeed {

--- a/website/docs/r/sentinel_data_connector_microsoft_threat_intelligence.html.markdown
+++ b/website/docs/r/sentinel_data_connector_microsoft_threat_intelligence.html.markdown
@@ -48,11 +48,7 @@ The following arguments are supported:
 
 * `log_analytics_workspace_id` - (Required) The ID of the Log Analytics Workspace. Changing this forces a new Data Connector to be created.
 
-* `microsoft_emerging_threat_feed_lookback_date` - (Optional) The lookback date for the Microsoft Emerging Threat Feed in RFC3339. Changing this forces a new Data Connector to be created.
-
--> **Note:** `microsoft_emerging_threat_feed_lookback_date` will be required in version 4.0 of the provider.
-
--> **NOTE:** At least one of `bing_safety_phishing_url_lookback_date` and `microsoft_emerging_threat_feed_lookback_date` must be specified.
+* `microsoft_emerging_threat_feed_lookback_date` - (Required) The lookback date for the Microsoft Emerging Threat Feed in RFC3339. Changing this forces a new Data Connector to be created.
 
 ---
 

